cocotb/share/lib/simulator/simulatormodule.c

```diff
@@ -219,7 +219,7 @@ static PyObject *register_readonly_callback(PyObject *self, PyObject *args)
     Py_ssize_t numargs = PyTuple_Size(args);
 
     if (numargs < 1) {
-        fprintf(stderr, "Attempt to register ReadOnly callback with!\n");
+        fprintf(stderr, "Attempt to register ReadOnly callback without enough arguments!\n");
```

Comment:
Note that if this line is reached, python will throw a `SystemError` shortly anyway. This needs to be `PyErr_SetString("...")`, as does almost every printf that appears before a `return NULL` - python functions are only allowed to return `NULL` if they set an exception